src/assets/passwordInput.js

```diff
@@ -0,0 +1,17 @@
+// Copyright © 2023 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+document.getElementsByName("password").forEach((p) => {
+  if (
+    p.type === "password" &&
+    !!(p.offsetWidth || p.offsetHeight || p.getClientRects().length)
```

Comment:
this checks if the element is in-fact in the DOM. 

How the JS logic works:
1. If there is a password input field and it is visibile (so the CSS did not hide it with `display: none`)
2. Then go through each input field with the name `password` and remove it if it is type `text` (custom input)
3. Else remove the current input field since it is not visible, but still type password. 

The CSS we have here does the following:
1. You are using a supported browser: show custom input field
2. You are not using a supported browser: hide custom input field

Fallback input is always of type `password` and custom is always of type `text`.